### PR TITLE
feat(theif): add latejoin theives

### PR DESCRIPTION
**Pull request by Brandon-Huu**
## About the PR

Adds a late-join mechanic for thieves 

## Why / Balance

image

## Technical details

bottom text

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

File: Content.Server/GameTicking/Rules/ThiefRuleSystem.cs [hunk 1]
```diff
@@ -101,6 +102,27 @@ public void MakeThief(EntityUid thief, ThiefRuleComponent thiefRule, bool addPac
...
+          continue;
+
+        if (!ev.LateJoin)
+          continue;
+        
+        if(!_antagSelection.IsPlayerEligible(ev.Player, comp.ThiefPrototypeId))
+          continue;
+        
+        if (_random.Prob(0.90f))
+          continue;
```
Comment (Brandon-Huu) [inline Content.Server/GameTicking/Rules/ThiefRuleSystem.cs:121]: When the maximum amount of thieves isn't achieved yet, the chance of randomly rolling thieves on late join is 10% based on this code. 

Should it be lowered? and if so? to what?

File: Content.Server/GameTicking/Rules/ThiefRuleSystem.cs [hunk 2]
```diff
@@ -101,6 +102,27 @@ public void MakeThief(EntityUid thief, ThiefRuleComponent thiefRule, bool addPac
...
         thiefRule.ThievesMinds.Add(mindId);
     }
+    
+    private void HandleLatejoin(PlayerSpawnCompleteEvent ev)
+    {
+      var query = QueryActiveRules();
+      while (query.MoveNext(out _, out var comp, out _))
+      {
+        if (comp.ThievesMinds.Count >= comp.MaxAllowThief)
+          continue;
```
Comment (Brandon-Huu) [inline Content.Server/GameTicking/Rules/ThiefRuleSystem.cs:112]: Is this good enough?

**Comment by metalgearsloth** [comment]
I'd rather wait for antag selection refactor to do this properly than have temporary code merged.

**Comment by Brandon-Huu** [comment]
> I'd rather wait for antag selection refactor to do this properly than have temporary code merged.

Bruh

**Pull request closed without merging**